.github/workflows/update-public-docs.yml

```diff
@@ -0,0 +1,78 @@
+---
+name: Update public docs
+
+on:
+  workflow_dispatch:
+  workflow_run:
+    workflows:
+      - CI
+    branches:
+      - main
+    status:
+      - completed
+
+jobs:
+  update-public-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: "actions/setup-python@v2"
+        with:
+          python-version: "3.9"
+      - uses: extractions/setup-just@v1
+
+      - name: Generate new public docs file
+        run: just generate-docs new_public_docs.json
+
+      - name: Clone docs repo
+        uses: actions/checkout@v3
+        with:
+          repository: opensafely/documentation
+          path: documentation
+          ref: main
+
+      - name: Check if docs have changed
+        id: check_docs
+        run: |
+          echo "::set-output name=DOCS_CHANGED::$(cmp new_public_docs.json documentation/public_docs.json)"
```

Comment:
You're right - I wrote the docs repo cloning/comparing for an earlier attempt at this and didn't think about comparing to a previous release.  And yes, it will create another release (and another docs PR) if the docs PR isn't merged quickly enough.  I did think about making the docs PRs automerge, but that seemed a bit riskly for the first stab at this.

I'll have a look at getting it to compare releases - I do think we're making enough changes on databuilder that the multiple releases could get out of hand

Reply:
@evansd I've updated it to check against the latest release, which was quite straightforward and makes the workflow faster since it doesn't have to checkout the docs repo now.

When I pushed a test branch though, I realised I need to do something with the build and publish workflow  too.  It [failed](https://github.com/opensafely-core/databuilder/runs/7355068285?check_suite_focus=true) - because it checks that the last tag was a version tag.  I'm not sure why that workflow is triggered on `create`, unless it was intended to run when a tag is created manually as well as via the CI workflow (it triggers on creation of a tag OR branch, which is why it ran when I created the test branch).  

I've changed the update_public_docs workflow so it triggers on completion of the build and publish one, rather than the CI one - I think that should mean that CI will run, tag a version if it needs to, trigger the build and publish workflow,, then trigger the update docs workflow - so a new release tag won't happen before a new version tag.  The problem (I think) will be if a new version wasn't required in the CI run; the build and publish workflow will still trigger, but instead of skipping the build steps, it'll fail because the last tag is a release tag.